File: Java/src/main/java/com/gildedrose/BackstagePass.java
package com.gildedrose;

public class BackstagePass extends ItemStrategy {
    @Override
    public void update(Item item) {
        if (item.quality < 50) {
            incrementQuality(item);
        }
        if (item.sellIn < 11) {
            if (item.quality < 50) {
                incrementQuality(item);
            }
        }
        if (item.sellIn < 6) {
            if (item.quality < 50) {
                incrementQuality(item);
            }
        }
        decrementSellin(item);
        if (item.sellIn < 0) {
            item.quality = item.quality - item.quality;
        }
    }
}

File: csharp/BackstagePass.cs
using System;
namespace GildedRose
{
    public class BackstagePass : ItemStrategy
    {
        public override void Update(Item item)
        {
            IncrementQuality(item);

            if (item.SellIn < 11)
            {

                IncrementQuality(item);

            }

            if (item.SellIn < 6)
            {

                IncrementQuality(item);
            }

            DecrementSellin(item);

            if (item.SellIn < 0)
            {
                item.Quality = item.Quality - item.Quality;

            }
        }
    }
}
